test/reports/ml_reports/test_counfounderAnalysis.py

```diff
@@ -0,0 +1,110 @@
+                                                                        implanting=GappedMotifImplanting())),
+                   Signal(identifier="age", motifs=[Motif(identifier="m3", instantiation=GappedKmerInstantiation(), seed="GGG")],
+                          implanting_strategy=HealthySequenceImplanting(implanting_computation=ImplantingComputation.ROUND,
+                                                                        implanting=GappedMotifImplanting()))]
+
+        simulation = Simulation([Implanting(dataset_implanting_rate=0.2, signals=signals, name='i1', repertoire_implanting_rate=0.25),
+                                 Implanting(dataset_implanting_rate=0.2, signals=[signals[0], signals[1]], name='i2', repertoire_implanting_rate=0.25),
+                                 Implanting(dataset_implanting_rate=0.1, signals=[signals[0]], name='i3', repertoire_implanting_rate=0.25),
+                                 Implanting(dataset_implanting_rate=0.2, signals=[signals[2]], name='i4', repertoire_implanting_rate=0.25),
+                                 Implanting(dataset_implanting_rate=0.1, signals=[signals[1]], name='i5', repertoire_implanting_rate=0.25)
+                                 ])
+
+        dataset = SignalImplanter.run(SimulationState(signals=signals, dataset=random_dataset, formats=['Pickle'], result_path=path,
+                                                      name='my_synthetic_dataset', simulation=simulation))
+
+        return dataset
+
+    def _encode_dataset(self, dataset, path, learn_model: bool = True):
+        encoder = KmerFrequencyEncoder.build_object(dataset, **{
+            "normalization_type": NormalizationType.RELATIVE_FREQUENCY.name,
+            "reads": ReadsType.UNIQUE.name,
+            "sequence_encoding": SequenceEncodingType.CONTINUOUS_KMER.name,
+            "k": 3
+        })  # encodes the repertoire by frequency of 3-mers
+        lc = LabelConfiguration()
+        lc.add_label("signal_disease", [True, False])
+        encoded_dataset = encoder.encode(dataset, EncoderParams(
+            result_path=path / "encoded",
+            label_config=lc,
+            learn_model=learn_model,
+            model={}
+        ))
+        return encoded_dataset
+
+    def _create_report(self, path):
+        report = ConfounderAnalysis.build_object(metadata_labels=["signal_age", "signal_HLA"])
+
+        report.ml_details_path = path / "ml_details.yaml"
+        report.label = "signal_disease"
+        report.result_path = path
+        report.train_dataset = self._encode_dataset(self._make_dataset(path / "train", size=100), path)
+        report.test_dataset = self._encode_dataset(self._make_dataset(path / "test", size=40), path, learn_model=False)
+        report.method = self._create_dummy_lr_model(path, report.train_dataset.encoded_data, report.label)
+
+        return report
+
+    def test_generate(self):
```

Comment:
Some things I didn't understand about the output/test:
- why are age and HLA a boolean value?
- if you have 'false positives' and 'false negatives' on the y axis, why is this a score from -1 to 1? I was instead expecting a number (count) or value between 0-100/0-1 (percentage). 
- at least in the current example, no data seems to be plotted

Reply:
Fixed the y-axis to be non-negative to avoid getting negative values on the y-axis when there are no misclassifications. It also only ticks at integer values.
Reduced the training set size for misclassifications to show up, to avoid empty plots.